sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs

```diff
@@ -42,30 +42,52 @@ public async override Task<AuthenticationEventResponse> Completed()
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    return await Failed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage)).ConfigureAwait(false);
                 }
                 else if (RequestStatus == RequestStatusType.TokenInvalid)
                 {
-                    Response.MarkAsUnauthorized();
+                    Unauthorized();
                 }
             }
             catch (Exception ex)
             {
-                return await Failed(ex, true).ConfigureAwait(false);
+                return await Failed(new AuthenticationEventTriggerRequestValidationException(ex.Message)).ConfigureAwait(false);
             }
 
             return Response;
         }
 
-        internal override Task<AuthenticationEventResponse> Failed(Exception exception, bool internalError)
+        /// <summary>
+        /// Sets the response statuscode, reason phrase and body when we want to return failed.
+        /// </summary>
+        /// <param name="exception">The exception that is thrown</param>
+        /// <returns>An authenticationeventresponse task</returns>
+        public override Task<AuthenticationEventResponse> Failed(Exception exception)
         {
             if (Response == null)
             {
                 Response = new TResponse();
             }
 
-            Response.MarkAsFailed(exception, internalError);
-            return Task.FromResult<AuthenticationEventResponse>((TResponse)Response);
+            Response.StatusCode = exception is AuthenticationEventTriggerValidationException ex ?
```

Comment:
Please group this into one if/else check

Reply:
done